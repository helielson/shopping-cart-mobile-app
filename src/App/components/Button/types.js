/**
 * Types definition used on Button Component
 *
 * @format
 * @flow
 */
// PropTypes for Button
export type ButtonProps = {
  withIcon?: boolean,
  iconName?: string,
  text: string,
  textColor?: string,
  onPress: () => mixed,
  otherProps?: {},
};
