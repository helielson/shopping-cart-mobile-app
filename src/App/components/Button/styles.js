import colors from '../../config/colors';

export default {
  button: {
    backgroundColor: colors.buttonColor,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 5,
    paddingHorizontal: 10,
    borderRadius: 5,
    flex: 1,
  },
  buttonText: {
    fontSize: 15,
    color: colors.backgroundLight,
  },
};
