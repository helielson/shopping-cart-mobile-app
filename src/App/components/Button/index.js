/**
 * Simple styled Button that can show a icon at at right of the text
 *
 * @param {ButtonProps} {
 *   withIcon,
 *   iconName,
 *   text,
 *   textColor,
 *   onPress,
 *   ...otherProps
 * }
 * @returns <Button />
 * @format
 * @flow
 */
import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import type { ButtonProps } from './types';
import styles from './styles';

const Button = ({ withIcon, iconName, text, textColor, onPress, ...otherProps }: ButtonProps) => (
  <TouchableOpacity {...otherProps} activeOpacity={0.6} onPress={onPress} style={styles.button}>
    {withIcon && <Icon name={iconName} size={24} color={textColor} />}
    <Text style={styles.buttonText}>{text}</Text>
  </TouchableOpacity>
);
/* Default Props for Button */
/* eslint-disable react/default-props-match-prop-types */
Button.defaultProps = {
  withIcon: false,
  iconName: 'add-shopping-cart',
  textColor: '#fff',
  otherProps: {},
};

// Exports component as default
export default Button;
