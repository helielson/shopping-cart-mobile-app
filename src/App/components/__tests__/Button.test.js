/**
 * Test Suite for Button Component
 *
 * @flow
 * @format
 * */
import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { shallow } from 'enzyme';
import { expect as chaiExpect } from 'chai';

import Button from '../Button';

describe('<Button />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Button withIcon text="Buy now!" onPress={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have one TouchableOpacity component', () => {
    const wrapper = shallow(<Button text="Buy now!" onPress={() => {}} />);
    chaiExpect(wrapper.find(TouchableOpacity)).to.have.length(1);
  });

  it('should have one Text component', () => {
    const wrapper = shallow(<Button text="Buy now!" onPress={() => {}} />);
    chaiExpect(wrapper.find(Text)).to.have.length(1);
  });

  it('should not have one Icon component', () => {
    const wrapper = shallow(<Button text="Buy now!" onPress={() => {}} />);
    chaiExpect(wrapper.find(Icon)).to.have.length(0);
  });

  it('should have one Icon component', () => {
    const wrapper = shallow(<Button withIcon text="Buy now!" onPress={() => {}} />);
    chaiExpect(wrapper.find(Icon)).to.have.length(1);
  });
});
