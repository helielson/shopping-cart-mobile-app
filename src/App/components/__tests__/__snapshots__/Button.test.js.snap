// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render properly 1`] = `
<TouchableOpacity
  activeOpacity={0.6}
  onPress={[Function]}
  otherProps={Object {}}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 145, 234, 1.00)",
      "borderRadius": 5,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <Icon
    allowFontScaling={false}
    color="#fff"
    name="add-shopping-cart"
    size={24}
  />
  <Text
    style={
      Object {
        "color": "rgba(245, 245, 246, 1.00)",
        "fontSize": 15,
      }
    }
  >
    Buy now!
  </Text>
</TouchableOpacity>
`;
